### PR TITLE
Simple chair script

### PR DESCRIPTION
**Pull request by PhilipRosedale**
Making a very simple chair script that can be used in tutorial and on chairs in marketplace. 

Need to check with avatar team on how to pin hips and make leaning better, as well as how to make avatar collisionless without flying animation, but this is good for now.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 88429c4033c73b0b27d167e15c17b1fe33538266 available at: https://gist.github.com/364d0c1ac2562e75d79df909a53b9c64<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9310-88429c4.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9310-88429c4.dmg

**Comment by PhilipRosedale** [comment]
Test plan:  

Attach this script to an entity whose center is roughly at waist level or a bit higher.   Set the entity to be non-colliding with avatars.  

Walk into it and notice that you should be snapped to the exact center of the entity, with your avatar rotated in the Z-direction of the entity, and your legs should assume a 'seated' pose.  

Walk away and the seated pose should end.  

Mouse-clicking at a distance on the entity should have the same effect.

**Comment by Nex-Pro** [comment]
###FAIL

Never needed to walk away. Seating pose cancelled every time after about one second when walked into or clicked.

Pose triggered and hips located in center.

Mouse clicking worked when "Triggerable" enabled. Annoyingly also triggers in edit mode.

**Comment by PhilipRosedale** [comment]
@Nex-Pro Fixed bug that caused it to cancel sitting like that.  Also added ability to sit by clicking with search beam.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0ad9a8625b23d0099b8d8caba7f4e477488cacae available at: https://gist.github.com/364d0c1ac2562e75d79df909a53b9c64<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9310-0ad9a86.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9310-0ad9a86.dmg

**Comment by PhilipRosedale** [comment]
OK works for Ryan and I... can't sit down if another person is in chair - give it a shot.  Agree about triggering or clicking while editing being annoying - submitted a bug on that.

**Comment by PhilipRosedale** [comment]
hang on... making a change for ryan

**Comment by hifi-gustavo** [comment]
Build console output for commit: 65b9aaed1b81458aded1310ff3fd7f9b4f799550 available at: https://gist.github.com/364d0c1ac2562e75d79df909a53b9c64<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9310-65b9aae.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9310-65b9aae.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a689d5200403b5fbc2509b276eefa8e603cbfeb2 available at: https://gist.github.com/364d0c1ac2562e75d79df909a53b9c64<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9310-a689d52.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9310-a689d52.dmg

File: scripts/tutorials/entity_scripts/chair.js [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+    var entity;
+    var props;
+    var checkTimer = false;
+    var settingsTimer = false;
+    var sitting = false;
+
+    var _this;
+    
+    var WANT_DEBUG = true;
```
Comment (ZappoMan) [inline scripts/tutorials/entity_scripts/chair.js]: please remove log spam - our CEO doesn't like it.

**Comment by BillWindwalker** [comment]
Is that chair script working ?
I removed all my hifi dir and did not think about backing up my scripts thinking it was server side.

**Pull request merged**